qualtran/bloqs/qft/qft_text_book_test.py

```diff
@@ -42,23 +44,28 @@ def test_qft_text_book(n: int, without_reverse: bool):
     assert np.allclose(cirq.unitary(qft_bloq), cirq.unitary(qft_cirq))
     assert np.allclose(cirq.unitary(qft_bloq**-1), cirq.unitary(qft_cirq**-1))
 
-    assert_valid_bloq_decomposition(qft_bloq)
+    qlt_testing.assert_valid_bloq_decomposition(qft_bloq)
 
 
 @pytest.mark.parametrize('n', [10, 123])
 def test_qft_text_book_t_complexity(n: int):
     qft_bloq = QFTTextBook(n)
-    qft_t_complexity = qft_bloq.t_complexity()
-    assert qft_t_complexity.rotations == (n * (n - 1)) // 2
-    assert qft_t_complexity.t == 0
+    qlt_testing.assert_equivalent_bloq_counts(qft_bloq, generalizer=[ignore_split_join])
+    gate_counts = get_cost_value(qft_bloq, QECGatesCost())
+    # special angle ZPow gets turned into clifford or T
+    rots = ((n - 3) * (n - 2)) // 2
```

Comment:
I don't know why this is failing specifically for `n=123` (and by a lot). There's (n-1) phase gradients, each of size `i` and we want to skip the first two (S, T). cc @tanujkhattar

Reply:
I didn't look at the individual ones. Do you mind taking a look?

Reply:
oooh interesting